app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt

```diff
@@ -0,0 +1,123 @@
+package com.robotemi.welcomingbtob.settings
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.Fragment
+import com.robotemi.sdk.Robot
+import com.robotemi.welcomingbtob.R
+import com.robotemi.welcomingbtob.widgets.CustomToggle
+import kotlinx.android.synthetic.main.fragment_configuration.*
+import org.koin.android.ext.android.inject
+
+class ConfigurationFragment : Fragment() {
+
+    private val robot: Robot by inject()
+
+    private val activityCallback by lazy { context as IActivityCallback }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_configuration, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        activityCallback.setTitle(getString(R.string.fragment_config))
+        activityCallback.setVisibilityOfDone(false)
+        activityCallback.setBackClickListener(View.OnClickListener { activity?.finish() })
+        updateUi(SettingsModel.getSettings(activity!!))
+        customToggleGreetUser.setToggleListener(object : CustomToggle.ToggleListener {
+            override fun onToggle(on: Boolean) {
+                val settings = getSettings()
+                settings.isUsingGreeterUser = on
+                if (!on) {
+                    settings.isUsingVoiceGreeter = false
+                }
+                saveSettings(settings)
+            }
+        })
+        customToggleVoiceGreeter.setToggleListener(object : CustomToggle.ToggleListener {
+            override fun onToggle(on: Boolean) {
+                val settings = getSettings()
+                settings.isUsingVoiceGreeter = on
+                saveSettings(settings)
```

Comment:
minimise to one line
saveSettings(getSettings().apply{isUsingVoiceGreeter = on})